src/main/java/org/assertj/core/api/AbstractClassAssert.java

```diff
@@ -284,4 +329,111 @@ public SELF hasDeclaredFields(String... fields) {
     classes.assertHasDeclaredFields(info, actual, fields);
     return myself;
   }
+
+  /**
+   * Verifies that the actual {@code Class} has the {@code methods}.
+   *
+   * <pre><code class='java'> class MyClass {
+   *     public void methodOne() {}
+   *     private void methodTwo() {}
+   * }
+   *
+   * // This one should pass :
+   * assertThat(MyClass.class).hasMethods("methodOne");
+   *
+   * // This one should fail :
+   * assertThat(MyClass.class).hasMethods("methodTwo");
+   * assertThat(MyClass.class).hasMethods("methodThree");</code></pre>
+   *
+   * @param methods the methods which must be in the class.
+   * @throws AssertionError if {@code actual} is {@code null}.
+   * @throws AssertionError if the actual {@code Class} doesn't contains all of the methods.
+   *
+   * @since 2.7.0 / 3.7.0
+   */
+  public SELF hasMethods(String... methods) {
+    classes.assertHasMethods(info, actual, methods);
+    return myself;
+  }
+
+  /**
+   * Verifies that the actual {@code Class} has the declared {@code methods}.
+   *
+   * <pre><code class='java'> class MyClass {
+   *     public void methodOne() {}
+   *     private void methodTwo() {}
+   * }
+   *
+   * // This one should pass :
+   * assertThat(MyClass.class).hasDeclaredMethods("methodOne", "methodTwo");
+   *
+   * // This one should fail :
+   * assertThat(MyClass.class).hasDeclaredMethods("methodThree");</code></pre>
```

Comment:
to make it clear, define a superclass for `MyClass` with a public method and use it in the failed assertion example (keep the example of the non existent method)